cylc/flow/option_parsers.py

```diff
@@ -181,10 +188,17 @@ def __init__(
             else:
                 argdoc = [('WORKFLOW', 'Workflow ID')]
 
-        if '--color=never' not in '='.join(sys.argv[2:]):
-            # Before option parsing, for `--help`, make comments grey in usage.
-            # (This catches both '--color=never' and '--color never'.)
+        joined_args = '='.join(sys.argv[2:])  # for "--opt=arg" and "--opt arg"
+        if (
+            '--color=never' not in joined_args
+            and
+            '--colour=never' not in joined_args
+        ):
+            # Before option parsing, make comments grey in --help output.
```

Comment:
This logic can be easily broken because the `--color` arg can appear at any position:

```bash
cylc --color=never
cylc config --color=never
cylc config foo --color=never
```

It can also take another format:

```bash
cylc config --color never
```

And won't get disabled automatically in non-interactive modes.

I think we can allow opt_parse to parse the color argument for us, but intercept the usage/description before they are printed using a help-formatter. We recently did something with help formatters in Rose:

https://github.com/metomi/rose/blob/3afc4d9c289433f6fc90f44e5b646ed880dc74d4/metomi/rose/opt_parse.py#L25-L37

I think this will do the trick:

```python
class CylcHelpFormatter(optparse.IndentedHelpFormatter):
    def _format(self, text):
        # note you can access cli opts here via self.parser.values
        if not self.parser.getattr('use_color'):
            text = strip(text)  # strip = AnsiMarkup().strip()
        return text

    def format_usage(self, usage):
        return super().format_usage(self._format(usage))

    def format_descrtiption(self, description):
        return super().format_description(self._format(description))


class CylcOptionParser(OptionParser):
    ...
    def __init__(self):
        ...
        kwargs['formatter'] = CylcHelpFormatter()
        OptionParser.__init__(self, *args, **kwargs)

    ...
    def parse_args(self):
        ...
        (options, args) = OptionParser.parse_args(self, api_args)                          
                                                                   
        self.use_color = (                                               
           opts.color == 'always'                                       
          or (opts.color == 'auto' and cylc.flow.terminal.supports_color())            
        )  
        color_init(autoreset=True, strip=not self.use_color)
```

(with the duplicated lines removed from `cylc.flow.terminal`).

Reply:
> This logic can be easily broken ... It can also take another format

Nope, the logic handles all of that. Proof (plus, see the tests I added):
![shot](https://user-images.githubusercontent.com/2362137/148447001-ae9de124-a694-4339-94d8-2065019af033.png)

It also works with the usage text:  `cylc scan --help | cat -` etc.